bridge_integration_test.go

```diff
@@ -1131,6 +1155,19 @@ func (m *mockRecorderClient) RecordToolUsage(ctx context.Context, req *aibridge.
 	return nil
 }
 
+// verify all interceptions has been marked as completed
+func (m *mockRecorderClient) verifyAllInterceptionsEnded(t *testing.T) {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+	if len(m.interceptions) == 0 {
+		t.Errorf("HMMM")
+	}
+	require.Equal(t, len(m.interceptions), len(m.interceptionsEnd))
+	for _, intc := range m.interceptions {
+		require.Contains(t, m.interceptionsEnd, intc.ID)
```

Comment:
Probably should add a message to this require call so it makes more sense what it's even doing. Perhaps also a `t.Helper()` call at the top of this method